diff --git a/build/src/start/server/metro/MetroBundlerDevServer.js b/build/src/start/server/metro/MetroBundlerDevServer.js
index f805cf4e0bed784e752248e8f9034a5e17c2f1a9..29090fa8fa2be1ac6aae95635538a1d6c1523ebd 100644
--- a/build/src/start/server/metro/MetroBundlerDevServer.js
+++ b/build/src/start/server/metro/MetroBundlerDevServer.js
@@ -207,7 +207,7 @@ class MetroBundlerDevServer extends _bundlerDevServer.BundlerDevServer {
             const contents = await this.bundleApiRoute(filepath, {
                 platform
             });
-            const artifactFilename = route.page === rscPath ? (0, _metroOptions.convertPathToModuleSpecifier)(_path().default.join(outputDir, "." + rscPath + ".js")) : (0, _metroOptions.convertPathToModuleSpecifier)(_path().default.join(outputDir, _path().default.relative(appDir, filepath.replace(/\.[tj]sx?$/, ".js"))));
+            const artifactFilename = route.page === rscPath ? (0, _metroOptions.convertPathToModuleSpecifier)(_path().default.join(outputDir, "." + rscPath + ".js")) : (0, _metroOptions.convertPathToModuleSpecifier)(_path().default.join(outputDir, _path().default.relative(appDir, filepath.replace(/\.[tj]sx?$/, ".cjs"))));
             if (contents) {
                 let src = contents.src;
                 if (includeSourceMaps && contents.map) {
