diff --git a/build/src/start/server/metro/MetroBundlerDevServer.js b/build/src/start/server/metro/MetroBundlerDevServer.js
index 9ad08175f726065726a11d0e07d46e1df767e835..d846e3740101bb3f5b50d29093321eeea6501282 100644
--- a/build/src/start/server/metro/MetroBundlerDevServer.js
+++ b/build/src/start/server/metro/MetroBundlerDevServer.js
@@ -207,7 +207,7 @@ class MetroBundlerDevServer extends _bundlerDevServer.BundlerDevServer {
             const contents = await this.bundleApiRoute(filepath, {
                 platform
             });
-            const artifactFilename = route.page === rscPath ? _path().default.join(outputDir, "." + rscPath + ".js") : _path().default.join(outputDir, _path().default.relative(appDir, filepath.replace(/\.[tj]sx?$/, ".js")));
+            const artifactFilename = route.page === rscPath ? _path().default.join(outputDir, "." + rscPath + ".js") : _path().default.join(outputDir, _path().default.relative(appDir, filepath.replace(/\.[tj]sx?$/, ".cjs")));
             if (contents) {
                 let src = contents.src;
                 if (includeSourceMaps && contents.map) {
