diff --git a/build/transform-worker/collect-dependencies.js b/build/transform-worker/collect-dependencies.js
index 85c77442fd1bd71bf531a6e51411241ad0ec5078..15949203660b0b3fe816660825d8be7af12d1a25 100644
--- a/build/transform-worker/collect-dependencies.js
+++ b/build/transform-worker/collect-dependencies.js
@@ -55,7 +55,7 @@ const debug = require('debug')('expo:metro:collect-dependencies');
 const MAGIC_IMPORT_COMMENTS = [
     '@metro-ignore',
     // Add support for Webpack ignore comment which is used in many different React libraries.
-    'webpackIgnore: true',
+    'webpackIgnore: true', 'webpackIgnore:true',
 ];
 // asserts non-null
 function nullthrows(x, message) {
