diff --git a/src/web/ImageWrapper.tsx b/src/web/ImageWrapper.tsx
index cb099db7e9eed5690aea4ae42178c4e4e230bbcf..de03f20553a2578750a7de19aac86a823349290d 100644
--- a/src/web/ImageWrapper.tsx
+++ b/src/web/ImageWrapper.tsx
@@ -44,7 +44,14 @@ const ImageWrapper = React.forwardRef(
       events?.onMount?.forEach((e) => e?.());
     }, []);
 
-    const tintId = useId();
+    // Use a unique ID for the SVG filter so that multiple Images can be used
+    // on the same page with different tint colors without conflicts.
+    const tintId = useId() 
+      // Make it safe for use as an SVG ID. SVG IDs are most strict than HTML IDs. They
+      // must be compliant with the https://www.w3.org/TR/xml/#NT-Name grammar. React 19
+      // changed the useId() implement return values that include « and ». Without this
+      // the SVG filter will not work in Safari which enforces the spec.
+      .replace(/[«»]/g, '_');
 
     // Thumbhash uri always has to start with 'thumbhash:/'
     const { resolvedSource, isImageHash } = useImageHashes(source);
