diff --git a/src/web/ColorTintFilter.tsx b/src/web/ColorTintFilter.tsx
index d8a3d6337da0b0afa3f83441e9cd9a61b744fba6..f0924b43233366effd11c65687dd40d567476234 100644
--- a/src/web/ColorTintFilter.tsx
+++ b/src/web/ColorTintFilter.tsx
@@ -1,25 +1,25 @@
 import React from 'react';
 import { StyleSheet } from 'react-native';
 
-export function getTintColorStyle(tintColor?: string | null) {
+export function getTintColorStyle(tintId: string, tintColor?: string | null) {
   if (!tintColor) {
     return {};
   }
   return {
-    filter: `url(#expo-image-tint-${tintColor})`,
+    filter: `url(#expo-image-tint-${tintId})`,
   };
 }
 
-type TintColorFilterProps = { tintColor?: string | null };
+type TintColorFilterProps = { id: string; tintColor?: string | null };  // https://github.com/expo/expo/pull/34604
 
-export default function TintColorFilter({ tintColor }: TintColorFilterProps) {
+export default function TintColorFilter({ id, tintColor }: TintColorFilterProps) {
   if (!tintColor) {
     return null;
   }
   return (
     <svg style={styles.svg}>
       <defs>
-        <filter id={`expo-image-tint-${tintColor}`}>
+        <filter id={`expo-image-tint-${id}`}>
           <feFlood floodColor={tintColor} />
           <feComposite in2="SourceAlpha" operator="atop" />
         </filter>
diff --git a/src/web/ImageWrapper.tsx b/src/web/ImageWrapper.tsx
index 31f35d5092359718002e087068dd8daa5a7f45f7..903648c05199eab3763450a58bb1d26398d23c3d 100644
--- a/src/web/ImageWrapper.tsx
+++ b/src/web/ImageWrapper.tsx
@@ -1,4 +1,4 @@
-import React, { useEffect, Ref } from 'react';
+import React, { useEffect, Ref, useId } from 'react';
 
 import ColorTintFilter, { getTintColorStyle } from './ColorTintFilter';
 import { ImageWrapperProps } from './ImageWrapper.types';
@@ -44,6 +44,8 @@ const ImageWrapper = React.forwardRef(
       events?.onMount?.forEach((e) => e?.());
     }, []);
 
+    const tintId = useId();
+
     // Thumbhash uri always has to start with 'thumbhash:/'
     const { resolvedSource, isImageHash } = useImageHashes(source);
     const objectPosition = getObjectPositionFromContentPositionObject(
@@ -56,7 +58,7 @@ const ImageWrapper = React.forwardRef(
     }
     return (
       <>
-        <ColorTintFilter tintColor={tintColor} />
+        <ColorTintFilter id={tintId} tintColor={tintColor} />
         <img
           ref={ref}
           alt={accessibilityLabel}
@@ -66,7 +68,7 @@ const ImageWrapper = React.forwardRef(
           style={{
             objectPosition,
             ...absoluteFilledPosition,
-            ...getTintColorStyle(tintColor),
+            ...getTintColorStyle(tintId, tintColor),
             ...style,
             ...(isImageHash ? hashPlaceholderStyle : {}),
           }}
