diff --git a/build/parse-keypress.js b/build/parse-keypress.js
index eec23c7ee003d22770cfedd37167edf3a50c5ee8..1e13e819f59d259a3476510db822695576a9d93c 100644
--- a/build/parse-keypress.js
+++ b/build/parse-keypress.js
@@ -175,7 +175,7 @@ const parseKeypress = (s = '') => {
         key.name = 'space';
         key.meta = s.length === 2;
     }
-    else if (s <= '\x1a') {
+    else if (s.length === 1 && s <= '\x1a') {
         // ctrl+letter
         key.name = String.fromCharCode(s.charCodeAt(0) + 'a'.charCodeAt(0) - 1);
         key.ctrl = true;
