diff --git a/dist/exports/Modal/ModalPortal.js b/dist/exports/Modal/ModalPortal.js
index df3bcbbd4dee13c614466ca30af52f10bf20398c..2af45b398714f88027b2748666ab8088973a3db4 100644
--- a/dist/exports/Modal/ModalPortal.js
+++ b/dist/exports/Modal/ModalPortal.js
@@ -15,22 +15,20 @@ function ModalPortal(props) {
   var children = props.children;
   var elementRef = React.useRef(null);
   if (canUseDOM && !elementRef.current) {
-    var element = document.createElement('div');
-    if (element && document.body) {
-      document.body.appendChild(element);
-      elementRef.current = element;
-    }
+    elementRef.current = document.createElement('div');
   }
-  React.useEffect(() => {
-    if (canUseDOM) {
+  React.useLayoutEffect(() => {
+    if (elementRef.current) {
+      if (!elementRef.current.parentElement) {
+        document.body.appendChild(elementRef.current);
+      }
       return () => {
-        if (document.body && elementRef.current) {
-          document.body.removeChild(elementRef.current);
-          elementRef.current = null;
+        if (elementRef.current.parentElement) {
+          elementRef.current.parentElement.removeChild(elementRef.current);
         }
       };
     }
-  }, []);
+  }, [elementRef.current])
   return elementRef.current && canUseDOM ? /*#__PURE__*/createPortal(children, elementRef.current) : null;
 }
 export default ModalPortal;
\ No newline at end of file
diff --git a/dist/exports/useColorScheme/index.js b/dist/exports/useColorScheme/index.js
index e917134c133eb7f438399d5fecf9543f3a49196a..a29aad7578667bcd03e972805a359f3f511d6921 100644
--- a/dist/exports/useColorScheme/index.js
+++ b/dist/exports/useColorScheme/index.js
@@ -12,17 +12,15 @@
 
 import * as React from 'react';
 import Appearance from '../Appearance';
+const subscribe = onStoreChange => {
+  const subscription = Appearance.addChangeListener(onStoreChange);
+  return subscription.remove;
+};
+const getServerSnapshot = () => 'light';
 export default function useColorScheme() {
-  var _React$useState = React.useState(Appearance.getColorScheme()),
-    colorScheme = _React$useState[0],
-    setColorScheme = _React$useState[1];
-  React.useEffect(() => {
-    function listener(appearance) {
-      setColorScheme(appearance.colorScheme);
-    }
-    var _Appearance$addChange = Appearance.addChangeListener(listener),
-      remove = _Appearance$addChange.remove;
-    return remove;
-  });
-  return colorScheme;
+  return React.useSyncExternalStore(
+    subscribe,
+    Appearance.getColorScheme,
+    getServerSnapshot
+  );
 }
\ No newline at end of file
