diff --git a/Libraries/Modal/Modal.d.ts b/Libraries/Modal/Modal.d.ts
index 21fca65479c1f1f9c54bd2703e502dff6ea02d96..43c9f064a37145f2ad1c1c6217db253554763fc0 100644
--- a/Libraries/Modal/Modal.d.ts
+++ b/Libraries/Modal/Modal.d.ts
@@ -7,10 +7,10 @@
  * @format
  */
 
-import type * as React from 'react';
-import {ViewProps} from '../Components/View/ViewPropTypes';
-import {NativeSyntheticEvent} from '../Types/CoreEventTypes';
-import {ColorValue} from '../StyleSheet/StyleSheet';
+import type * as React from "react";
+import { ViewProps } from "../Components/View/ViewPropTypes";
+import { NativeSyntheticEvent } from "../Types/CoreEventTypes";
+import { ColorValue } from "../StyleSheet/StyleSheet";
 
 export interface ModalBaseProps {
   /**
@@ -24,7 +24,7 @@ export interface ModalBaseProps {
    * - `fade` fades into view
    * - `none` appears without an animation
    */
-  animationType?: 'none' | 'slide' | 'fade' | undefined;
+  animationType?: "none" | "slide" | "fade" | undefined;
   /**
    * The `transparent` prop determines whether your modal will fill the entire view.
    * Setting this to `true` will render the modal over a transparent background.
@@ -49,7 +49,7 @@ export interface ModalBaseProps {
    * The `backdropColor` props sets the background color of the modal's container.
    * Defaults to `white` if not provided and transparent is `false`. Ignored if `transparent` is `true`.
    */
-  backdropColor?: ColorValue | undefined;
+  backdropColor?: ColorValue | false | undefined;
 }
 
 export interface ModalPropsIOS {
@@ -57,10 +57,10 @@ export interface ModalPropsIOS {
    * The `presentationStyle` determines the style of modal to show
    */
   presentationStyle?:
-    | 'fullScreen'
-    | 'pageSheet'
-    | 'formSheet'
-    | 'overFullScreen'
+    | "fullScreen"
+    | "pageSheet"
+    | "formSheet"
+    | "overFullScreen"
     | undefined;
 
   /**
@@ -69,11 +69,11 @@ export interface ModalPropsIOS {
    */
   supportedOrientations?:
     | Array<
-        | 'portrait'
-        | 'portrait-upside-down'
-        | 'landscape'
-        | 'landscape-left'
-        | 'landscape-right'
+        | "portrait"
+        | "portrait-upside-down"
+        | "landscape"
+        | "landscape-left"
+        | "landscape-right"
       >
     | undefined;
 
