diff --git a/out/chunk-C1TWiB5U.d.ts b/out/chunk-C1TWiB5U.d.ts
index 7d41352eb4442f1b40a7eb6ca4edfe108f34bfec..8285c8f2db0b3b6b201495e8111434fa615fc265 100644
--- a/out/chunk-C1TWiB5U.d.ts
+++ b/out/chunk-C1TWiB5U.d.ts
@@ -6,7 +6,7 @@ type JSONValue =
   | string
   | boolean
   | number
-  | Array<JSONValue>
+  | Array<unknown>
   | JSONObject;
 
 /**
@@ -31,12 +31,12 @@ type ReadonlyJSONValue =
   | string
   | boolean
   | number
-  | ReadonlyArray<ReadonlyJSONValue>
+  | ReadonlyArray<unknown>
   | ReadonlyJSONObject;
 
 /** Like {@link JSONObject} but deeply readonly */
 type ReadonlyJSONObject = {
-  readonly [key: string]: ReadonlyJSONValue | undefined;
+  readonly [key: string]: unknown | undefined;
 };
 
 type MaybePromise<T> = T | Promise<T>;
