diff --git a/lib/lib.dom.d.ts b/lib/lib.dom.d.ts
index edf900a5de22fd63305f87e32d5974716eda7ddd..5424fbf6e03778151b444b9907f35aab0e773370 100644
--- a/lib/lib.dom.d.ts
+++ b/lib/lib.dom.d.ts
@@ -19901,7 +19901,7 @@ declare var Response: {
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/error_static) */
     error(): Response;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/json_static) */
-    json(data: any, init?: ResponseInit): Response;
+    json(data: string | object, init?: ResponseInit): Response;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/redirect_static) */
     redirect(url: string | URL, status?: number): Response;
 };
