diff --git a/lib/lib.dom.d.ts b/lib/lib.dom.d.ts
index 14508e8dc3effb02a4bd808e61770aacbc8b13b7..f4d19ebe21c537e6ace41617e8eb89667046192b 100644
--- a/lib/lib.dom.d.ts
+++ b/lib/lib.dom.d.ts
@@ -26378,7 +26378,7 @@ declare var Response: {
      *
      * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Response/json_static)
      */
-    json(data: any, init?: ResponseInit): Response;
+    json(data: object | string, init?: ResponseInit): Response;
     /**
      * The **`redirect()`** static method of the Response interface returns a `Response` resulting in a redirect to the specified URL.
      *
@@ -36749,6 +36749,7 @@ interface Window extends EventTarget, AnimationFrameProvider, GlobalEventHandler
     releaseEvents(): void;
     /**
      * The **`window.requestIdleCallback()`** method queues a function to be called during a browser's idle periods.
+     * @deprecated not supported on Safari
      *
      * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/requestIdleCallback)
      */
@@ -38785,6 +38786,7 @@ declare function prompt(message?: string, _default?: string): string | null;
 declare function releaseEvents(): void;
 /**
  * The **`window.requestIdleCallback()`** method queues a function to be called during a browser's idle periods.
+ * @deprecated not supported on Safari
  *
  * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Window/requestIdleCallback)
  */
diff --git a/lib/typescript.js b/lib/typescript.js
index 2643aa12aa6497ca0b90ecd99b202f37ea0e6330..5c165de6fd31e0eee2043e17aa4d6472bec51fde 100644
--- a/lib/typescript.js
+++ b/lib/typescript.js
@@ -146297,7 +146297,7 @@ function getFunctionOrClassName(node) {
       if (name.length > maxLength) {
         return `${name} callback`;
       }
-      const args = cleanText(mapDefined(parent2.arguments, (a) => isStringLiteralLike(a) || isTemplateLiteral(a) ? a.getText(curSourceFile) : void 0).join(", "));
+      const args = cleanText(mapDefined(parent2.arguments, (a) => { if (isSatisfiesExpression(a)) { a = a.expression; } return isStringLiteralLike(a) || isTemplateLiteral(a) ? a.getText(curSourceFile) : void 0; }).join(", "));
       return `${name}(${args}) callback`;
     }
   }
